stdlib/number.gr

```diff
@@ -398,3 +398,77 @@ export let sin = (radians: Number) => {
 export let cos = (radians: Number) => {
   sin(pi / 2 + radians)
 }
+
+/**
+ * Computes the tan of a number (in radians) using Chebyshev polynomials.
+ *
+ * @param radians: The input in radians
+ * @returns The computed tangent
+ *
+ * @since v0.5.2
+ */
+export let tan = (radians: Number) => {
+  sin(radians) / cos(radians)
+}
+
+/**
+ * Computes the factorial of a number.
+ *
+ * @param input: The number to compute the factorial of
+ * @returns The computed factorial
+ *
+ * @since v0.5.2
+ */
+export let factorial = input => {
+  let s = sign(input)
+  let mut result = 1
+  for (let mut n = abs(input); n > 0; n -= 1) {
+    result *= n
+  }
+  if (s == 0) result else result * s
+}
+
+/**
+ * Clamps a number to a given range.
+ *
+ * @param min: The minimum value in the range
+ * @param max: The maximum value in the range
+ * @param input: The number to clamp
+ * @returns The clamped number
+ *
+ * @since v0.5.2
+ */
+export let clamp = (min: Number, max: Number, input: Number) => {
+  if (input < min) min else if (input > max) max else input
+}
+
+/**
+ * Linearly interpolates between the starting and ending value by the given amount.
+ *
+ * @param start: The starting value
+ * @param end: The ending value
+ * @param amount: an amount between 0 and 1
+ * @returns The interpolated value
+ *
+ * @since v0.5.2
+ */
+export let lerp = (start: Number, end: Number, amount: Number) => {
+  (end - start) * amount + start
+}
+
+/**
+ * Maps the given number from one range to another.
+ *
+ * @param inMin: The minimum value in the input range
+ * @param outMin: The minimum value in the output range
+ * @param inMax: The maximum value in the input range
+ * @param outMax: The maximum value in the output range
+ * @param current: The number to map
+ * @returns The interpolated value
+ *
+ * @since v0.5.2
+ */
+export let map = (inMin, outMin, inMax, outMax, current) => {
```

Comment:
I think this name would confuse a lot of FP folks. Maybe a name like `linearMap` would be better?

Reply:
could we reexport it as that like an alias because I think `linearMap` is going to confuse a lot of oop people and cause them to use a local implementation

Reply:
Definitely want it to only have one name. Since Grain is an FP language we cater a bit more to the FP use case 🙂 

`map` might be a fine name; we'll see what some others think when they review.